### PR TITLE
[feature][Connector-V2][cdc] Support flink running cdc job

### LINKED ISSUE #4605: [Feature][CDC] Support flink running cdc job
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Support flink running cdc job

### validate

Remove `@ DisabledOnContainer ` from  `SqlServerCDCIT `
https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-sqlserver-e2e/src/test/java/org/apache/seatunnel/e2e/connector/cdc/sqlserver/SqlServerCDCIT.java#L63

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Carl-Zhou-CN**
## Purpose of this pull request

close #4605

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
good

**Comment by TyrantLucifer** [comment]
good pr

**Comment by EricJoy2048** [comment]
niubility!

**Comment by hailin0** [comment]
waitting for ci check

**Comment by Carl-Zhou-CN** [comment]
@hailin0 Please help to start the client,the mysql cdc driver is abnormal

**Comment by Carl-Zhou-CN** [comment]
image
Appears to be a zeta startup exception

**Comment by hailin0** [comment]
@EricJoy2048

**Comment by evan766** [comment]
> ## Purpose of this pull request
> ## Check list
> * [ ]  Code changed are covered with tests, or it does not need tests for reason:
> * [ ]  If any new Jar binary package adding in your PR, please add License Notice according
>   [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
> * [ ]  If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
> * [ ]  If you are contributing the connector code, please check that the following files are updated:
>   
>   1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
>   2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
>   3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
> * [ ]  Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).


兄弟，我clone了你的项目，用的flink-cdc分支编译，再创建mysql-cdc任务到 flink,  flink还是报错，你这边是OK的吗

**Comment by Carl-Zhou-CN** [comment]
@evan766 Can you share your json configuration?

**Comment by evan766** [comment]
> @evan766 Can you share your json configuration?

thank you very much to reply me!
steps I do, where was wrong?
1、fork your project: https://github.com/Carl-Zhou-CN/incubator-seatunnel/tree/flink-cdc
2、here is my forked project: `https://github.com/evan766/incubator-seatunnel`
3、clone https://github.com/evan766/incubator-seatunnel to my local machine.
4 、
```
git clone git@github.com:evan766/incubator-seatunnel.git
checkout  flink-cdc
checkout -b  dev-cdc
git remote add apache-seatunnel https://github.com/apache/seatunnel.git
git merge apache-seatunnel/dev
git push origin
```
5、follow the steps in the tutorial to build binanry package： https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/setup.md
```
#Install Subproject Locally
./mvnw install -Dmaven.test.skip

#Building seaTunnel from source
mvn clean package -pl seatunnel-dist -am -Dmaven.test.skip=true
```

6、terminal  output  target in dir:  incubator-seatunnel/seatunnel-dist/target
```
$: ls -l incubator-seatunnel/seatunnel-dist/target

-rw-r--r--@ 1 hong  staff  1593852848 Jul 12 11:56 apache-seatunnel-2.3.2-SNAPSHOT-bin.tar.gz
-rw-r--r--@ 1 hong  staff     4838534 Jul 12 11:56 apache-seatunnel-2.3.2-SNAPSHOT-src.tar.gz
drwxr-xr-x@ 2 hong  staff          64 Jul 12 11:55 archive-tmp
drwxr-xr-x@ 3 hong  staff          96 Jul 12 11:55 maven-shared-archive-resources
-rw-r--r--@ 1 hong  staff          81 Jul 12 11:55 spotless-index

```
7、 push job to flink , flink Version: 1.16.2 
```
tar zxvf apache-seatunnel-2.3.2-SNAPSHOT-bin.tar.gz
bash bin/start-seatunnel-flink-15-connector-v2.sh --config config/cdc.template
```
cdc.templte
```
env {
  execution.parallelism = 1
  job.mode = "STREAMING"
  checkpoint.interval = 10000
}

source {
  MySQL-CDC {
  debezium {
          poll.interval.ms = 1000
      }
    result_table_name = "temp_organization"
    parallelism = 1
    server-id = 5656
    username = "root"
    password = "root"
    table-names = ["testbeat.organization"]
    base-url = "jdbc:mysql://192.168.66.227:3306/testbeat"
  }
}

sink {
    jdbc {
          source_table_name:"temp_organization"
          url = "jdbc:mysql://192.168.66.227:3306/seatunnel"
          driver = "com.mysql.cj.jdbc.Driver"
          user = "root"
          password = "root"
          generate_sink_sql = "true"
          database = "seatunnel"
          table = "organization2"
          primary_keys = ["id"]
    }
}
```
8、flink job failed:

```
2023-07-12 14:10:02
java.lang.UnsupportedOperationException: Flink ParallelSource don't support sending SourceEvent. Please implement the `SupportCoordinate` marker interface on the SeaTunnel source.
	at org.apache.seatunnel.translation.source.ParallelReaderContext.sendSourceEventToEnumerator(ParallelReaderContext.java:61)
	at org.apache.seatunnel.connectors.cdc.base.source.reader.IncrementalSourceReader.reportFinishedSnapshotSplitsIfNeed(IncrementalSourceReader.java:151)
	at org.apache.seatunnel.connectors.cdc.base.source.reader.IncrementalSourceReader.onSplitFinished(IncrementalSourceReader.java:133)
	at org.apache.seatunnel.connectors.seatunnel.common.source.reader.SourceReaderBase.finishCurrentFetch(SourceReaderBase.java:204)
	at org.apache.seatunnel.connectors.seatunnel.common.source.reader.SourceReaderBase.moveToNextSplit(SourceReaderBase.java:180)
	at org.apache.seatunnel.connectors.seatunnel.common.source.reader.SourceReaderBase.getNextFetch(SourceReaderBase.java:161)
	at org.apache.seatunnel.connectors.seatunnel.common.source.reader.SourceReaderBase.pollNext(SourceReaderBase.java:92)
	at org.apache.seatunnel.connectors.cdc.base.source.reader.IncrementalSourceReader.pollNext(IncrementalSourceReader.java:92)
	at org.apache.seatunnel.connectors.seatunnel.common.source.reader.SourceReaderBase.pollNext(SourceReaderBase.java:112)
	at org.apache.seatunnel.connectors.cdc.base.source.reader.IncrementalSourceReader.pollNext(IncrementalSourceReader.java:92)
	at org.apache.seatunnel.translation.source.ParallelSource.run(ParallelSource.java:136)
	at org.apache.seatunnel.translation.flink.source.BaseSeaTunnelSourceFunction.run(BaseSeaTunnelSourceFunction.java:86)
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:110)
	at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:67)
	at org.apache.flink.streaming.runtime.tasks.SourceStreamTask$LegacySourceFunctionThread.run(SourceStreamTask.java:333)

```
image

**Comment by Carl-Zhou-CN** [comment]
image
Whether this was removed when merged？@evan766

**Comment by evan766** [comment]
i check the code , is not removed. could you please clone&nbsp; the code from my github repo:&nbsp;https://github.com/evan766/incubator-seatunnel.git, the merged branch is:&nbsp;dev-cdc




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年7月12日(星期三) 下午2:51
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [apache/seatunnel] [feature][Connector-V2][cdc] Support flink running cdc job (PR #4918)





 

 Whether this was removed when merged？
 
&mdash;

**Comment by Carl-Zhou-CN** [comment]
@evan766 Can I add my wechat zy330600

**Comment by hailin0** [comment]
Please merge dev branch into this pr

@Carl-Zhou-CN

**Comment by Carl-Zhou-CN** [comment]
@hailin0 ok

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-sqlserver-e2e/src/test/java/org/apache/seatunnel/e2e/connector/cdc/sqlserver/SqlServerCDCIT.java [hunk 1]
```diff
@@ -65,7 +65,7 @@
 @Slf4j
 @DisabledOnContainer(
         value = {},
-        type = {EngineType.SPARK, EngineType.FLINK},
+        type = {EngineType.SPARK},
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-sqlserver-e2e/src/test/java/org/apache/seatunnel/e2e/connector/cdc/sqlserver/SqlServerCDCIT.java:68]: update testcase

https://github.com/apache/seatunnel/blob/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mongodb-e2e/src/test/java/mongodb/MongodbCDCIT.java#L68
https://github.com/apache/seatunnel/blob/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java#L58

**Comment by hailin0** [comment]
check ci error

**Comment by Carl-Zhou-CN** [comment]
Temporarily blocks mongo cdc effects #5251

**Comment by Carl-Zhou-CN** [comment]
image
In his own warehouse he was successful

**Comment by Carl-Zhou-CN** [comment]
@ic4y @hailin0 @EricJoy2048 @Hisoka-X @TyrantLucifer  Please help review, thank you

**Comment by hailin0** [comment]
LGTM

**Review by hailin0 — APPROVED** [review]

**Review by TyrantLucifer — APPROVED** [review]
LGTM, thank you for your contribution!

File: seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/FlinkAbstractPluginExecuteProcessor.java [hunk 2]
```diff
@@ -76,14 +80,41 @@ protected Optional<DataStream<Row>> fromSourceTable(Config pluginConfig) {
...
+                        resultTable,
+                        isAppendMap.getOrDefault(sourceTable, true));
+                registerAppendStream(pluginConfig);
+                return;
+            }
+            flinkRuntimeEnvironment.registerResultTable(
+                    pluginConfig,
+                    dataStream,
+                    resultTable,
+                    isAppendMap.getOrDefault(resultTable, true));
```
Comment (Hisoka-X) [inline seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/FlinkAbstractPluginExecuteProcessor.java:109]: ~~Hi @Carl-Zhou-CN , could you explain for me why we should register table as source table name when both have `RESULT_TABLE_NAME` and `SOURCE_TABLE_NAME`? Thanks.~~

I got it. Please ignore it.

**Pull request merged**